### PR TITLE
Get started page UX improvements: add get started with video card

### PR DESCRIPTION
**Pull request by ianlin**
### Changes proposed in this Pull Request:

Part of #1392, this PR add the first card of the new get started page. The first card includes the same marketing video in the [Google Listings & Ads WooCommerce product page](https://woocommerce.com/products/google-listings-and-ads/).

To be able to play the video, I referred to the implementation from 10517-gh-Automattic/woocommerce.com but using [Wistia's JavaScript Player API](https://wistia.com/support/developers/player-api#with-iframe-embeds) to get the video state in order to expand the video height after the user clicks on the video.

To be able to use Wistia's API, I add a custom hook `useScript` and use it to load Wistia's script dynamically in the component.

Note that the video will not be expanded in the mobile version.

Design for desktop: eQ9O4m2flzrcAiDMXkOW0m-fi-7350%3A120914
Design for mobile: eQ9O4m2flzrcAiDMXkOW0m-fi-7476%3A122828

#### Brave (Chromium)

https://user-images.githubusercontent.com/914406/163765567-62a6b434-8359-4f07-b492-4df607bb9085.mov

### Firefox

https://user-images.githubusercontent.com/914406/163766064-938019f3-b02b-4183-96e8-248e9575ae3c.mov

### Safari

https://user-images.githubusercontent.com/914406/163767314-cdf16c48-8f95-4dd5-a4e6-6c111308d767.mov

### Detailed test instructions:

1. Go to GLA's get started page: `https://domain.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart`
2. Check the first card of the get started page, the UI should align with the design.
3. The video will be expanded when playing the video on the desktop.
4. The video won't be expanded when playing the video on the mobile.
5. Click on the `Start listing products` and it will go to the set up page.

### Changelog entry

> Update - Add the first card with a CTA and a video for UX improvements on get started page.

### CODE REVIEW

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+/**
+ * Internal dependencies
+ */
+import { glaData } from '.~/constants';
+import AppButton from '.~/components/app-button';
+import AppDocumentationLink from '.~/components/app-documentation-link';
+import WistiaVideo from '.~/components/wistia-video';
+import footerImageURL from './footer.svg';
+import googleImageURL from './google.svg';
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❔ 
There is already a Google logo at [here](https://github.com/woocommerce/google-listings-and-ads/blob/0cfd8f16e7cd061e974c62cb3430b38c0918b380/js/src/product-feed/submission-success-guide/google-logo.svg). It could be reused by setting the needed image width or height.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import { getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { glaData } from '.~/constants';
+import AppButton from '.~/components/app-button';
+import AppDocumentationLink from '.~/components/app-documentation-link';
+import WistiaVideo from '.~/components/wistia-video';
+import footerImageURL from './footer.svg';
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❓ 
This icon and the right-side description could be replaced with \<Tip\> provided by `@wordpress/components`.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import googleImageURL from './google.svg';
+import './index.scss';
+
+const GetStartedWithVideoCard = () => {
+	const disableNextStep = ! glaData.mcSupportedLanguage;
+
+	return (
+		<Card
+			className="woocommerce-marketing-google-get-started-with-video-card"
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❔ 
I guess the *woocommerce-marketing-googl* prefix was only used at the beginning of this plugin, but it's too long then and didn't be used other than the Get started page. Currently, most CSS class names are started with *gla-*.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+			isBorderless
+		>
+			<Flex>
+				<FlexBlock className="header">
+					<Text variant="body" className="title">
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❓ 
Suggest avoiding using CSS names that easily name collisions.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+						{ __(
+							'The official extension for WooCommerce, built in collaboration with',
+							'google-listings-and-ads'
+						) }
+					</Text>
+					<div id="google-img">
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❓ 
Suggest using `className` instead because `id` usually is only adopted when there is a concrete reason. From the perspective of component design, most components are not designed with the premise of unique existing on the whole page, even though sometimes they are the only instance in the runtime.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+						{ __(
+							'Sync your products directly to Google, manage your product feed, and create Google Ad campaigns–without leaving your WooCommerce dashboard. The official extension, built in collaboration with Google.',
+							'google-listings-and-ads'
+						) }
+					</Text>
+					<AppButton
+						className="button"
+						isPrimary
+						disabled={ disableNextStep }
+						href={ getNewPath( {}, '/google/setup-mc' ) }
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: 💅  It could be replaced with https://github.com/woocommerce/google-listings-and-ads/blob/0cfd8f16e7cd061e974c62cb3430b38c0918b380/js/src/utils/urls.js#L50-L52

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+						eventName="gla_setup_mc"
+						eventProps={ {
+							target: 'set_up_free_listings',
+							trigger: 'click',
+						} }
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: 💅  Suggest adding event JSDoc for auto-generating the tracking docs.

https://github.com/woocommerce/google-listings-and-ads/blob/0cfd8f16e7cd061e974c62cb3430b38c0918b380/js/src/get-started-page/get-started-card/index.js#L24-L26

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+					>
+						{ __(
+							'Start listing products →',
+							'google-listings-and-ads'
+						) }
+					</AppButton>
+					<Text className="hint">
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❓  Same concern for the CSS naming collisions.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+										context="get-started"
+										linkId="wp-terms-of-service"
+										href="https://wordpress.com/tos/"
+									/>
+								),
+							}
+						) }
+					</Text>
+				</FlexBlock>
+				<FlexBlock className="footer">
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❓ Same concern for the CSS naming collisions.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+						) }
+					</Text>
+					<Text variant="body" className="description">
+						{ __(
+							'Sync your products directly to Google, manage your product feed, and create Google Ad campaigns–without leaving your WooCommerce dashboard. The official extension, built in collaboration with Google.',
+							'google-listings-and-ads'
+						) }
+					</Text>
+					<AppButton
+						className="button"
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❓ Same concern for the CSS naming collisions.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+										href="https://wordpress.com/tos/"
+									/>
+								),
+							}
+						) }
+					</Text>
+				</FlexBlock>
+				<FlexBlock className="footer">
+					<img src={ footerImageURL } alt="" />
+					<Text variant="body" className="hint">
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: ❓ Same concern for the CSS naming collisions.

File: js/src/get-started-page/get-started-with-video-card/index.js [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+						} }
+					>
+						{ __(
+							'Start listing products →',
+							'google-listings-and-ads'
+						) }
+					</AppButton>
+					<Text className="hint">
+						{ __(
+							'Estimated setup time: 15 min ',
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.js]: 💅  The space char at the end could be removed.

File: js/src/get-started-page/get-started-with-video-card/index.scss [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+				.title {
+					padding: 0;
+					font-size: 20px;
+					line-height: $default-line-height;
+				}
+			}
+
+			.footer {
+				margin: $grid-unit-20;
+			}
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.scss]: 💅 
According to the visual design, the icon is aligned to the top in the mobile view.
image

File: js/src/get-started-page/get-started-with-video-card/index.scss [hunk 14]
```diff
@@ -0,0 +1,120 @@
+.woocommerce-marketing-google-get-started-with-video-card {
+	.components-flex {
+		flex-direction: column;
+
+		.header {
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.scss]: 💅  or 📜 
With the CardHeader, CardBody, and CardFooter components provided by `@wordpress/components`, I think some styles in this file could be omitted and be more consistent with the codebase.

File: js/src/get-started-page/index.js [hunk 15]
```diff
@@ -5,12 +5,14 @@ import './index.scss';
...
 import GetStartedCard from './get-started-card';
+import GetStartedWithVideoCard from './get-started-with-video-card';
 import FeaturesCard from './features-card';
 
 const GetStartedPage = () => {
 	return (
 		<div className="woocommerce-marketing-google-get-started-page">
 			<UnsupportedNotices />
+			<GetStartedWithVideoCard />
 			<GetStartedCard />
```
Comment (eason9487) [inline js/src/get-started-page/index.js:16]: ❔ or 💅 
Would the GetStartedCard components be removed in another PR, or we could remove them along with this PR?

File: js/src/hooks/useScript.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Load an external script and call the provided callback function when the script is loaded.
+ *
+ * @param {string}   url      The script's url.
+ * @param {Function} callback A function to be called after the script is loaded.
+ */
+const useScript = ( url, callback = () => {} ) => {
```
Comment (eason9487) [inline js/src/hooks/useScript.js]: 💅  The default value of `callback` is not needed if it's a required parameter. Or, if the `callback` is optional, it'd be better to indicate that in the JSDoc and use the same reference for the default value. Otherwise, the callback of `useEffect` will be triggered when every time rendering.

Usually. The *noop* can be imported from `lodash` or declared at the module scope.

```js
import { noop } from 'lodash';
// or
const noop = () => {};
```

File: js/src/components/wistia-video/index.js [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+	useEffect( () => {
+		if ( wistiaVideoHandle ) {
+			const intervalId = setInterval( () => {
+				if ( wistiaVideoHandle.state() === 'playing' ) {
+					setPlaying( true );
+					clearInterval( intervalId );
+				}
+			}, 100 );
+		}
+	}, [ wistiaVideoHandle ] );
```
Comment (eason9487) [inline js/src/components/wistia-video/index.js]: 💅  This handling could be replaced with the following in the callback of `useScript`:
```js
window._wq.push( {
	id,
	onReady: ( video ) => {
		video.bind( 'play', () => setPlaying( true ) );
	},
} );
```

File: js/src/components/wistia-video/index.js [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+	useScript( WISTIA_API_SCRIPT_URL, () => {
+		if ( window._wq ) {
+			window._wq.push( {
+				id,
+				onReady: ( video ) => {
+					setWistaVideoHandle( video );
+				},
+			} );
+		}
+	} );
```
Comment (eason9487) [inline js/src/components/wistia-video/index.js]: 🚧 
The callback function is a different reference in each rendering, which leads the script might be appended to and removed from the document body several times, and the initial processing callback itself is called several times. Even though it appears that wistia internally handles the problem of script reloading and reinitialization, this still added some potential concerns. Suggestions to avoid unexpected changes in the callback reference.

Kapture 2022-04-18 at 19 33 35

File: js/src/components/wistia-video/index.scss [hunk 19]
```diff
@@ -0,0 +1,43 @@
+.wistia-responsive-padding {
```
Comment (eason9487) [inline js/src/components/wistia-video/index.scss]: ❔ Wondering why it's named with `-padding` postfix?

File: js/src/components/wistia-video/index.js [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+	};
+
+	return (
+		<div
+			className={ classNames(
+				'wistia-responsive-padding',
+				isPlaying ? 'is-playing' : ''
+			) }
+		>
+			<div className="wistia-responsive-wrapper">
```
Comment (eason9487) [inline js/src/components/wistia-video/index.js]: 💅
In GLA codebase, the naming convention of subcomponent is usually named in the form of gla-main-component-name__part-name. Using __ as a separator makes the main and sub distinguishable.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 1 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks, updated in 40c6663.

File: js/src/get-started-page/index.js (hunk 15 above)
Comment (ianlin) [reply js/src/get-started-page/index.js:16]: Thanks for the reminder, there is a new card get `GetStartedCard` as well like in the design: eQ9O4m2flzrcAiDMXkOW0m-fi-7350%3A121253, so I will update it in another PR.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 4 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Refactored the css part in 6eb1e68 and 0aba591.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 5 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Refactored the css part in 6eb1e68 and 0aba591.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 8 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Refactored the css part in 6eb1e68 and 0aba591.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 9 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Refactored the css part in 6eb1e68 and 0aba591.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 10 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Refactored the css part in 6eb1e68 and 0aba591.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 11 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Refactored the css part in 6eb1e68 and 0aba591.

File: js/src/get-started-page/get-started-with-video-card/index.scss (hunk 14 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.scss]: Thanks! Refactored the css part in 6eb1e68 and 0aba591.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 2 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Updated in 5759c51.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 3 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Updated in e84172b.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 6 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Updated in 1b95995.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 7 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: Thanks! Added in b0f903d.

File: js/src/get-started-page/get-started-with-video-card/index.js (hunk 12 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.js]: 👀  Thanks! Removed in dadeb91.

File: js/src/get-started-page/get-started-with-video-card/index.scss (hunk 13 above)
Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.scss]: Thanks! Updated in df9220a.

File: js/src/hooks/useScript.js (hunk 16 above)
Comment (ianlin) [reply js/src/hooks/useScript.js]: Thanks! Updated in c426ee5.

File: js/src/components/wistia-video/index.js (hunk 17 above)
Comment (ianlin) [reply js/src/components/wistia-video/index.js]: Awesome! Thanks for the suggestion, updated in 71a9913.

File: js/src/components/wistia-video/index.js (hunk 18 above)
Comment (ianlin) [reply js/src/components/wistia-video/index.js]: Thanks for spotting this! Updated in 910e112 and confirmed it won't be loading the script again when re-render.

File: js/src/components/wistia-video/index.scss (hunk 19 above)
Comment (ianlin) [reply js/src/components/wistia-video/index.scss]: It was copied from the Wistia's sample code 😅. Updated in 2c31424, thanks!

File: js/src/components/wistia-video/index.js (hunk 20 above)
Comment (ianlin) [reply js/src/components/wistia-video/index.js]: Thanks, also updated in 2c31424.

**Comment by ianlin** [comment]
Hey @eason9487! Thanks for all the review, it's super useful. I've updated the code based on your suggestions and it's ready for another round of the review.

File: js/src/get-started-page/get-started-with-video-card/index.scss [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+	}
+
+	// Repeat selector to make it higher priority.
+	& &__description {
+		margin-bottom: $grid-unit-40;
+		padding: 0 $grid-unit-10;
+	}
+
+	// Repeat selector to make it higher priority.
+	& &__button {
```
Comment (eason9487) [inline js/src/get-started-page/get-started-with-video-card/index.scss]: 💅 

Some styling details could be considered.
- The padding of `.components-card__header` still has a lower priority than `@wordpress/components`.
- Specify the proper content height to avoid overriding the spacing and hardcoded height on the container. This also brings a flexible layout when the content changes or is dynamic.
- Some CSS styles could be omitted.
- Remove some code comments since `& &__` is not a repeated-selector pattern.

I added a code suggestion block because it might be clearer than using a lot of comments. 🙏 
```suggestion
	// Adjust <CardHeader> imported from @wordpress/components.
	// Repeat selector to make it higher priority.
	.components-card__header.components-card__header {
		justify-content: flex-end;
		padding: $grid-unit * 2.5;

		& > p {
			margin: 0 1em;
			line-height: 16px;
			font-size: $helptext-font-size;
			color: $gray-600;
		}

		@media (max-width: $break-small) {
			justify-content: center;
			text-align: center;
			padding: $grid-unit-20;

			& > p::after {
				content: " Google";
			}

			& > img {
				display: none;
			}
		}
	}

	.motivation-video {
		min-width: 100%;
		margin: 0;
	}

	// Adjust <CardBody> imported from @wordpress/components.
	// Repeat selector to make it higher priority.
	.components-card__body.components-card__body {
		display: flex;
		flex-direction: column;
		align-items: center;
		text-align: center;
		margin: calc(var(--main-gap) * 1.67) calc(var(--main-gap) * 4.67);
		padding: 0;

		@media (max-width: $break-small) {
			margin: $grid-unit-30;
		}
	}

	& &__title {
		padding: 0 $grid-unit-30;
		margin-bottom: $grid-unit-15;
	}

	& &__description {
		margin-bottom: $grid-unit-40;
		padding: 0 $grid-unit-10;
	}

	& &__button {
```

**Review by eason9487 — APPROVED** [review]
Thanks for the updates! Testing well. LGTM. Left a comment about the styling, and it's not a blocker.

File: js/src/get-started-page/get-started-with-video-card/index.scss (hunk 21 above)
Comment (eason9487) [reply js/src/get-started-page/get-started-with-video-card/index.scss]: I missed the other two code comments below, which are also removable.

Comment (ianlin) [reply js/src/get-started-page/get-started-with-video-card/index.scss]: Thanks for the suggested changes @eason9487 , it's much more cleaner now! I've got a question regarding:

> & &__ is not a repeated-selector pattern.

Agree it's different from the repeating form like `className.className`. But using `& &__` actually solved the priority problem to me in this case. I checked the [MDN CSS Specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#how_is_specificity_calculated) doc and it seems like `& &__` matches either the rule of

1. Make better use of the CSS cascade, or
2. Use more specific rules. By indicating one or more elements before the element you're selecting, the rule becomes more specific and gets higher priority

So it could have higher priority to override the `Text` component's style from `@wordpress/components`. I'd like to add some comments like `// Use more specific rules to make it higher priority.` to those use `& &__`, what do you think about it?

Comment (eason9487) [reply js/src/get-started-page/get-started-with-video-card/index.scss]: > So it could have higher priority to override the `Text` component's style from `@wordpress/components`. I'd like to add some comments like `// Use more specific rules to make it higher priority.` to those use `& &__`, what do you think about it?

It might seem a bit strange at first glance to see something like `& &__title`, but the purpose of this syntax becomes clear after thinking about the compiled result like `.card .card__title`. And its purpose is exactly the either or both reasons you mentioned. Since these two purposes are very common rules for writing/reading CSS, they usually don't need to be commented on.

But on the other hand, when styling for a component that is entirely under our control, there is no need to override some other styles, then the use of `&__title` is usually sufficient. So even if the reason for these selectors can be deduced here is to raise the priority over `Text`, it would be still fine to leave comments for this concrete reason.
